src/components/Pagination/Pagination.tsx

```diff
@@ -144,11 +160,124 @@ const Pagination = (props: PaginationProps) => {
         : value.total)
     );
   }
+  function onPaginationRange() {
+    // Check if the media query is true
+    const widths = [window.innerWidth];
+    if (window.screen?.width) {
+      widths.push(window.screen?.width);
+    }
+    const width = Math.min(...widths);
+
+    if (width <= 768) {
+      const paginationRange = usePagination(
+        4,
+        value.total,
+        value.limit,
+        1,
+        value.current
+      );
+      setPaginationRange(paginationRange);
+    } else {
+      const paginationRange = usePagination(
+        5,
+        value.total,
+        value.limit,
+        2,
+        value.current
+      );
+      setPaginationRange(paginationRange);
+    }
+  }
+  function displayPages() {
+    return paginationRange?.map((i: any, index: any) => (
+      <span
+        key={index}
+        id={index}
+        onClick={(e) => selectedPage(e, i, index)}
+        className={`n-pagination__number_inactive ${
+          i === value.current && "n-pagination__number_active"
+        } ${i === "..." && openPopup && "n-pagination__dot_active"}`}
+      >
+        {i}
+      </span>
+    ));
+  }
+  function selectedPage(e: any, i: any, index: any) {
+    if (i == "...") {
+      const ele = document.getElementById(index);
+      const rect = ele && ele.getBoundingClientRect();
+
+      const x = rect?.left;
+      const y = rect?.top;
+      let popupData = document.getElementById("menu");
+      popupData && (popupData.style.top = `${y}px`);
+      popupData && (popupData.style.left = `${x}px`);
```

Comment:
Avoid inline styles, please add dynamic classes as necessary. Inline styles will make it hard to track css